assets/questions/palindrome.js

```diff
@@ -0,0 +1,152 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Question data for Valid Palindrome.
+ */
+
+globalData.questions['palindrome'] = {  // eslint-disable-line dot-notation
+  title: 'Balanced Palindrome',
+  starterCode: {
+    python:
+`def isPalindrome(s):
+    return False
+`
+  },
+  auxiliaryCode: {
+    python:
+`class AuxiliaryCode(object):
+    @classmethod
+    def createPalindromeString(cls, chr_set, input_size):
+        import random
+        ret = ""
+        while input_size > 0:
+            chr = random.choice(chr_set)
+            if input_size == 1:
+                ret += chr
+                input_size -= 1
+            else:
+                ret = chr + ret + chr
+                input_size -= 2
+        return ret
+
+    @classmethod
+    def countNumberOfChar(cls, s):
+        ascii_codes = {i:0 for i in range(255)}
+        for chr in s:
+            ascii_codes[ord(chr)] += 1
+        for code in ascii_codes:
+            if ascii_codes[code] % 2 is not 0:
+                return False
+        return True
+
+    @classmethod
+    def forgetIgnoreSpace(cls, s):
+        for i in range(len(s)/2):
+            if s[i] is not s[-i-1]:
+                return False
+        return True
+`
+  },
+  tasks: [{
+    instructions: [
+      [
+        'For this question, you will implement the isPalindrom function. ',
+        'It takes a sentense as input and return True if the input is a ',
+        'palindrome and False if it is not.'
+      ].join(''),
+      [
+        'A palindrom is a word, phrase, or sequence that reads the same ',
+        'backward as forward, e.g., \'madam\' or \'nurses run\'.'
+      ].join(''),
+      'Note, the input string would only consist of alphabets from a-z and space.'
```

Comment:
Note that the input string will consist of lowercase letters (a - z) and spaces.